lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2923,53 +2930,71 @@ void SplitEmitter::emitModule(const LoweringOptions &options,
   auto op = mod.op;
 
   // Given the operation, determine the file stem name and how to emit it.
-  std::function<void(VerilogEmitterState &)> emit;
+  std::function<void(VerilogEmitterState &)> emitWithinFileOps;
+  std::function<void(llvm::raw_ostream & os)> emit =
+      [&](llvm::raw_ostream &os) {
```

Comment:
When you split the new IR construct out to its own patch, please add a self contained testcase for it.